cmd/zed/zed_event.c

```diff
@@ -100,6 +100,45 @@ zed_event_fini(struct zed_conf *zcp)
 	zed_exec_fini();
 }
 
+static void
+_bump_event_queue_length(void)
+{
+	int zzlm = -1, wr;
+	char qlen_buf[12] = {0}; /* parameter is int => max "-2147483647\n" */
+	long int qlen;
+
+	zzlm = open("/sys/module/zfs/parameters/zfs_zevent_len_max", O_RDWR);
+	if (zzlm < 0)
+		goto done;
+
+	if (read(zzlm, qlen_buf, sizeof (qlen_buf)) < 0)
+		goto done;
+	qlen_buf[sizeof (qlen_buf) - 1] = '\0';
+
+	errno = 0;
+	qlen = strtol(qlen_buf, NULL, 10);
+	if (errno == ERANGE)
+		goto done;
+
+	if (qlen <= 0)
+		qlen = 512; /* as good a value as any, really */
```

Comment:
`512` is a good minimum since it matched the default value of `zfs_zevent_len_max` as of commit 032a213e2e6b791364301e9de029e7b8189a69a0.  I'd update the comment to, "default zfs_zevent_len_max value`.  Or something like that.

Reply:
Oh, if it does then it's a happy coincidence – it was 64 on my 2-CPU system and 384(?) on my 24-CPU system and the manpage said it was autoadjusted, so I picked something Slightly Bigger.